File: HotelReservationServiceJava/Reservation.java

public class Reservation
{
    
    private static int idCount = 1;
    
    String clientName;
    int id;
    int arrival;
    int duration;//days
    int personCount;
    Room room;
    
    public Reservation(String clientName, int arrival, int duration, int personCount){
        this.clientName = clientName;
        this.arrival = arrival;
        this.duration = duration;
        this.personCount = personCount;
        
        this.id = idCount++;
    }
    
    public String getClientName(){
        return clientName;
    }
    
    public void setRoom(Room room){
        this.room = room;
    }
    
    public int getArrival(){
        return arrival;
    }
    
    public int getDuration(){
        return duration;
    }
    
    public int getPersonCount(){
        return personCount;
    }
    
    public int getId(){
        return id;
    }
    
    public Room getRoom(){
        return room;
    }
    
}

File: HotelReservationServiceCpp/src/Reservation.cpp
#include "Reservation.h"

int Reservation::idCount = 1;

Reservation::Reservation(std::string cName, int arr, int dur, int pCount) :
    clientName{cName},
    arrival{arr},
    duration{dur},
    personCount{pCount} {
        this->id = idCount++;
}

int Reservation::getId(){
    return this->id;
}

void Reservation::setRoom(Room& room){
    this->room = &room;
}

int Reservation::getArrival(){
    return this->arrival;
}

std::string Reservation::getClientName(){
    return clientName;
}

int Reservation::getDuration(){
    return this->duration;
}

Room * Reservation::getRoom(){
    return this->room;
}

int Reservation::getPersonCount(){
    return this->personCount;
}

Reservation::~Reservation() {}
